autoload/intero/process.vim

```diff
@@ -67,7 +72,9 @@ function! intero#process#initialize() abort
             call s:start_compile(10, l:opts)
         else
             let g:intero_built = 1
-            call intero#process#start()
+            if g:intero_start_immediately
```

Comment:
I think this check should actually be in `after/ftplugin/haskell/intero.vim` - otherwise when you open a project it will compile Intero even if `intero_start_immediately` is set to 0. (And if you think intero is heavy, that's going to be way worse)